tmk_core/common/chibios/pin_defs.h

```diff
@@ -53,6 +53,9 @@
 #        define D5 PAL_LINE(GPIOC, 13)
 #        define B0 PAL_LINE(GPIOC, 14)
 #    endif
+#elif defined(KEYBOARD_kinesis_kint36)
+// The compatibility defines do not work with the KINETIS header files,
+// so do not declare anything.
```

Comment:
```suggestion
```

Reply:
I’m a little confused about this suggestion—just removing these lines will result in a compilation error down the line (likely only once #10171 is applied).

Are you saying we should defer this change until later?

Reply:
There needs to be a better solution than just hardcoding keyboard names into core code whenever someone wants to add a Teensy 3.6-based board.

What is the error it's producing?

Reply:
> There needs to be a better solution than just hardcoding keyboard names into core code whenever someone wants to add a Teensy 3.6-based board.

Definitely agreed. If you can tell me how to make a MK66F18 define work, that’d be great!

> What is the error it's producing?

Here’s the full compilation error: https://gist.github.com/stapelberg/260314ef7f6310d3c54527c73e6442b0

Reply:
I think this is due to the config_common.h include in the top-level config.h - this is also why the Input Club boards don't include that file. Considering the top-level one doesn't reference any of the pin definitions, moving the include to each of the sub-config.h's would probably "solve" it.

Reply:
Ah, thanks for figuring this out! Yes, I think you are right.

We can change the config includes in #10171